### PR TITLE
DROID-2398 Multiplayer | Enhancement | Allow picking an image from device as space icon

### PR DESCRIPTION
**Pull request by uburoiubu**

### CODE REVIEW

File: presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt [hunk 1]
```diff
@@ -325,6 +330,41 @@ class SpaceSettingsViewModel(
         }
     }
 
+    fun onSpaceImagePicked(path: String) {
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt:333]: pls add Timber.d

File: presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt [hunk 2]
```diff
@@ -325,6 +330,41 @@ class SpaceSettingsViewModel(
...
+        viewModelScope.launch {
+            uploadFile.async(
+                params = UploadFile.Params(
+                    path = path,
+                    space = params.space,
+                    type = Block.Content.File.Type.IMAGE
+                )
+            ).fold(
+                onSuccess = { file ->
+                    setSpaceDetails.async(
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt:343]: better to move setSpaceDetails to an another function

File: ui-settings/src/main/java/com/anytypeio/anytype/ui_settings/main/MainSettingScreen.kt [hunk 3]
```diff
@@ -34,8 +43,18 @@ fun SpaceHeader(
...
     onRandomGradientClicked: () -> Unit,
-    isEditEnabled: Boolean
+    isEditEnabled: Boolean,
+    onSpaceImagePicked: (Uri) -> Unit
 ) {
+    val context = LocalContext.current
+    val singlePhotoPickerLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.PickVisualMedia(),
+        onResult = { uri ->
+            if (uri != null) {
```
Comment (konstantiniiv) [inline ui-settings/src/main/java/com/anytypeio/anytype/ui_settings/main/MainSettingScreen.kt:53]: else Timber.e

**Review by konstantiniiv — APPROVED** [review]

File: ui-settings/src/main/java/com/anytypeio/anytype/ui_settings/main/MainSettingScreen.kt (hunk 3 above)
Comment (uburoiubu) [reply ui-settings/src/main/java/com/anytypeio/anytype/ui_settings/main/MainSettingScreen.kt:53]: done

File: presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt (hunk 2 above)
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt:343]: done

File: presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt (hunk 1 above)
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceSettingsViewModel.kt:333]: done

**Pull request merged**